app/presenters/search_result_presenter.rb

```diff
@@ -8,29 +10,76 @@ class SearchResultPresenter
            :es_score,
            to: :search_result
 
-  def initialize(search_result, metadata)
-    @search_result = search_result
-    @metadata = metadata
+  def initialize(data = {})
+    @search_result = data[:search_result]
+    @metadata = data[:metadata]
+    @index = data[:doc_index] + 1
+    @count = data[:doc_count]
+    @finder_name = data[:finder_name]
+    @debug_score = data[:debug_score]
+    @highlight = data[:highlight]
   end
 
   def to_hash
     {
-      title: title,
-      link: link,
-      summary: summary,
-      is_historic: is_historic,
-      government_name: government_name,
-      metadata: metadata,
-      show_metadata: show_metadata,
-      format: format,
-      es_score: es_score,
+      link: {
+        text: title,
+        path: link,
+        description: summary_text,
+        data_attributes: {
+          track_category: "navFinderLinkClicked",
+          track_action: "#{@finder_name}.#{@index}",
+          track_label: link,
+          track_options: {
+            dimension28: @count,
+            dimension29: title
+          }
+        }
+      },
+      metadata: structure_metadata,
+      metadata_raw: metadata,
```

Comment:
I assume metadata_raw isn't being used by the component? Maybe it is better to expose the raw metadata in a separate method instead of being part of this hash?

Reply:
Correct. It's used by the grouped result presenter. I would like to do this in a different way, but it seems entangled to quite a degree, if you look at the code. Any suggestions welcome, but I wonder if this is something bigger we should tackle in a separate PR?